src/lightning/fabric/fabric.py

```diff
@@ -584,9 +584,12 @@ def no_backward_sync(self, module: _FabricModule, enabled: bool = True) -> Gener
 
     @contextmanager
     def sharded_model(self) -> Generator:
-        """Shard the parameters of the model instantly when instantiating the layers."""
-        context = self._strategy.init_sharded_context() if isinstance(self._strategy, _Sharded) else nullcontext()
-        with context:
+        """Instantiate a model under this context manager to prepare it for model-parallel sharding.
+
+        .. deprecated:: This context manager is deprecated in favor of :meth:`init_module`, use it instead.
+        """
+        rank_zero_deprecation("`Fabric.sharded_model()` is deprecated in favor of `Fabric.init_module()`.")
```

Comment:
Since we agree on adding back this deprecation, it could be pulled out into a separate PR to get it merged sooner

Reply:
Please review this: https://github.com/Lightning-AI/lightning/pull/17721

Reply:
Done!